theano/compile/tests/test_function_module.py

```diff
@@ -239,6 +239,41 @@ def test_copy(self):
         f(1, 2)  # put them out of sync
         self.assertFalse(f(1, 2) == g(1, 2))  # they should not be equal anymore.
 
+    def test_copy_share_memory(self):
+        x = T.fscalar('x')
+        y = T.tanh((x+2)/(x-0.2)**2)
+
+        # test for PerformaLinker, will cover VM_linker later
+        ori = theano.function([x], [y], mode="FAST_COMPILE")
```

Comment:
I think the copy isn't needed. But in all cases, we don't need to uptimise this case as DebugMode handle the memory differently for verification. So not testing DebugMode is good.